src/main/java/mate/academy/mybookshop/service/BookServiceImpl.java

```diff
@@ -46,11 +53,29 @@ public BookDto update(Long id, UpdateBookRequestDto bookRequestDto) {
 
         var updatedBook = bookMapper.toEntity(bookRequestDto);
         updatedBook.setId(id);
+        setCategories(updatedBook, bookRequestDto.getCategoryIds());
         return bookMapper.toDto(bookRepository.save(updatedBook));
     }
 
     @Override
     public void delete(Long id) {
         bookRepository.deleteById(id);
     }
+
+    @Override
+    public List<BookDtoWithoutCategoryIds> findAllBooksByCategoryId(
+            Long categoryId, Pageable pageable) {
+        List<BookEntity> bookEntities = bookRepository.findAllByCategoriesId(categoryId, pageable);
+        return bookEntities.stream()
+                .map(bookMapper::toDtoWithoutCategories)
+                .toList();
+    }
+
+    private void setCategories(BookEntity bookEntity, Set<Long> categoryIds) {
+        if (categoryIds != null && !categoryIds.isEmpty()) {
+            Set<CategoryEntity> categories = new HashSet<>(
+                    categoryRepository.findAllById(categoryIds));
```

Comment:
waht if some of those categories that we received from user do not exist? we should throw an exception in that case

Reply:
added